autospec/infile_bb_parser.py

```diff
@@ -0,0 +1,202 @@
+#!/usr/bin/true
+#
+# infile_parser.py - part of autospec
+# Copyright (C) 2017 Intel Corporation
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+import re
+
+# TODO: VAR_append = "" _append same functionality as +=
+operators_dict = {
+    "=": "{1}",
+    "=.": "{1}{0}",
+    ".=": "{0}{1}",
+    ":=": "{1}",
+    "=+": "{1} {0}",
+    "+=": "{0} {1}",
+    "??=": "{1}",
+    "?=": "{1}"
+}
+
+
+def scrape_version(bb_dict):
+    # get version from latest filename added to dict, if exists
+    try:
+        bb_dict['VERSION'] = bb_dict.get('filename')[-1].split('_', 1,)[1].rsplit('.', 1)[0]
+    except IndexError:
+        print('file has no version: {}')
+
+    return bb_dict
+
+
+def replace_pv(bb_dict):
+    for k, v in bb_dict.items():
+        if "${PV}" in v and "VERSION" in bb_dict:
+            v = v.replace("${PV}", bb_dict.get('VERSION'))
+        if "${ROS_SPN}" in v:
+            if "ROS_SPN" in bb_dict:
+                v = v.replace("${ROS_SPN}", bb_dict.get("ROS_SPN"))
+            elif "NAME" in bb_dict:
+                v = v.replace("${ROS_SPN}", bb_dict.get('NAME'))
+
+        bb_dict[k] = v
+
+
+def get_src_url(bb_dict):
+    if bb_dict.get('SRC_URI'):
+        bb_dict["URL"] = bb_dict.get("SRC_URI").split('\\')[0].split(';')[0]
+
+
+def clean_values(value):
+    # remove beginning and end quote
+    if value.startswith('"') and value.endswith('"'):
+        value = value[1:-1]
+
+    # remove line continuations
+    value = "".join(value.split("\\"))
+    return value
+
+
+def update_inherit(line, bb_dict):
+    if 'inherits' in bb_dict:
+        bb_dict['inherits'].append(' '.join(line.split(' ', 1)[1:]))
```

Comment:
inherits can be multiple lines, that always start with "inherit", but can have multiple values following it. For example:
inherit autotools update-alternatives
inherit autotools-brokensep
I was splitting them to remove the inherit, and then store as a list, with each line its own value. I assume order matters, so it would be bb_dict["inherit"] = ['autotools' 'update-alternatives', 'autotools-brokensep']. I'll do some research to see if they can each be stored as their own value in a list (splitting autotools and update-alternatives to be 2 values, not 1)

Reply:
Inherits can have multiple lines, and multiple values on each line: 
inherit autotools update-alternatives
inherit autotools-brokensep

Currently its storing a list with ["autotools update-alternatives", "autotools-brokensep"] as opposed to each inherit being its own item. My reasoning for updating is so that the value doesn't get over written if there is more than one inherit.